src/samples/coap/oic-client.c

```diff
@@ -111,6 +111,15 @@ found_resource(struct sol_oic_client *cli, struct sol_oic_resource *res, void *d
     if (!res)
         return false;
 
+#ifndef SOL_NO_API_VERSION
```

Comment:
This is a callback that gives you the type, no need to check it. 

Same for others in this file